routing/payment_lifecycle.go

```diff
@@ -37,7 +37,7 @@ type paymentLifecycle struct {
 	finalCLTVDelta uint16
 	attempt        *channeldb.PaymentAttemptInfo
 	circuit        *sphinx.Circuit
-	lastError      error
+	lastError      *htlcswitch.ForwardingError
```

Comment:
should be done for `errNoRoute` error also. Can also squash the two commits together.

Reply:
yes, good point. fixed and squashed.